tests/ecosystem/upgrade/conftest.py

```diff
@@ -0,0 +1,276 @@
+import logging
+import threading
+
+import pytest
+
+from ocs_ci.ocs import constants
+
+log = logging.getLogger(__name__)
+
+
+def create_pods(
+    interface,
+    pvc_factory,
+    pod_factory,
+    storageclass,
+    count,
+    access_mode,
+    volume_mode=None
+):
+    """
+    Create pods for upgrade testing. pvc_factory and pod_factory have to be
+    in the same scope.
+
+    Args:
+        interface (str): CephBlockPool or CephFileSystem
+        pvc_factory (function): Function for creating PVCs
+        pod_factory (function): Function for creating pods
+        storageclass (obj): Storageclass to use
+        count (int): Number of pods to create
+        access_mode (str): ReadWriteOnce, ReadOnlyMany or ReadWriteMany.
+            This decides the access mode to be used for the PVC
+        volume_mode (str): Volume mode for rbd RWO PVC
+
+    Return:
+        list: List of generated pods
+    """
+    log.info(
+        f"Creating {count} pods via {interface} using {access_mode}"
+        f" access mode, {volume_mode} volume mode and {storageclass.name}"
+        f" storageclass"
+    )
+    pvcs = [
+        pvc_factory(
+            storageclass=storageclass,
+            access_mode=access_mode,
+            volume_mode=volume_mode
+        ) for _ in range(count)
+    ]
+    if volume_mode == constants.VOLUME_MODE_BLOCK:
+        pod_dict = constants.CSI_RBD_RAW_BLOCK_POD_YAML
+        raw_block_pv = True
+    else:
+        pod_dict = ''
+        raw_block_pv = False
+    pods = [
+        pod_factory(
+            interface=interface,
+            pvc=pvc,
+            raw_block_pv=raw_block_pv,
+            pod_dict_path=pod_dict,
+        ) for pvc in pvcs
+    ]
+    return pods
+
+
+@pytest.fixture(scope='session')
+def pre_upgrade_filesystem_pods(
+    request,
```

Comment:
@fbalak Looks really good!
Please consider using https://github.com/red-hat-storage/ocs-ci/blob/dcf872c69993d39b77a4efe97d02d36f40ce9775/tests/conftest.py#L791.
It's a session scoped fixture will take care of both pods creation on block and file and run IO in BG.

Reply:
Thanks, in future it would be good to merge this fixture and workload from this PR as this PR covers more configurations including Retain Reclaim Policy and different access and volume modes. For now I would keep it separate as there is created a lot of pods for the utilization during upgrade and it would be good plan and decide if this is a kind of workload that we want to run during execution of all test cases.

Reply:
There is also created ten pods for each configuration which can be quite a big number for normal test run.

Reply:
@fbalak thanks for looking into this.
Currently, run_io_in_bg fixture is not being used at all. 
Feel free to modify it so it will create 10 pods for each configuration